src/sagas/console/index.js

```diff
@@ -175,11 +166,20 @@ export function* openConsoleModal (action) {
 // ----- Console Options (per VM) held by `OptionsManager`
 export function* getConsoleOptions (action) {
   const options = OptionsManager.loadConsoleOptions(action.payload)
-  yield put(setConsoleOptions({ vmId: action.payload.vmId, options }))
+  yield put(Actions.setConsoleOptions({ vmId: action.payload.vmId, options }))
   return options
 }
 
 export function* saveConsoleOptions (action) {
   OptionsManager.saveConsoleOptions(action.payload)
-  yield getConsoleOptions(getConsoleOptionsAction({ vmId: action.payload.vmId }))
+  yield getConsoleOptions(Actions.getConsoleOptions({ vmId: action.payload.vmId }))
+}
+
+export function* autoConnect (vmId) {
+  console.warn('inside autoconnect', vmId)
+  const vm = yield call(fetchSingleVm, Actions.getSingleVm({ vmId, shallowFetch: true }))
+  console.warn('autoconnect fetch', vm)
+  if (vm && vm.id && vm.status !== 'down') {
+    yield downloadVmConsole(Actions.downloadConsole({ vmId, hasGuestAgent: vm.ssoGuestAgent }))
```

Comment:
`downloadVmConsole()` is a misleading name these days.  Maybe rename that function to `downloadOrOpenVmConsole()` since it will acutally push a use to the novnc console if things are configured that way...